doc/overlays.xml

```diff
@@ -0,0 +1,99 @@
+<chapter xmlns="http://docbook.org/ns/docbook"
+         xmlns:xlink="http://www.w3.org/1999/xlink"
+         xml:id="chap-overlays">
+
+<title>Overlays</title>
+
+<para>This chapter describes how to extend and change Nixpkgs content using
+overlays.  Overlays are used to add layers in the fix-point used by Nixpkgs
+to bind the dependencies of all packages.</para>
```

Comment:
Overlays can also be used for adding packages, so I would make this a bit more general, maybe `used by Nixpkgs to compose the set of all packages`.